### DISCUSSION TITLE
how to get immundeconv cibersort to match standalone cibersort

### DISCUSSION THREAD

**Discussion by RichardCorbett** [omnideconv/immunedeconv#138] [Q&A]
Hi all,

I thought this would be simple, but so far the solution has eluded me.

I have STAR/RSEM ens100 genes.results files.    When I pull out the HGNC and TPM columns and supply them to standalone CIBERSORT (in absolute mode) I get a CD8+ score of 0.07.

When I load my genes.results files into R and then run CIBERSORT through immunedeconv I get a CD8+ estimate of 0.14.

in my standalone CIBERSORT analysis I use the following params:
```
CIBERSORT(LM22_signature_file.txt, test_prod_mixture_file.txt , perm=500, QN=FALSE, absolute=TRUE, abs_method="sig.score")
```

and then through immunedeconv I use the following line:
```
res = deconvolute(TPMs, "cibersort_abs", abs_method = "sig.score",  QN=FALSE, perm=500)
```

I've been at this for days and haven't found the cause of the difference.   The gene list, LM22, and TPM values are the same going into both commands.     I'm sure I'm missing something obvious.   Has anyone done this themselves and figured it out?

**Comment by LorenzoMerotto** [comment]
Hello @RichardCorbett 
I think the problem is related to the quantile normalization. 
The  functions in immunedeconv have a uinfied set of parameter names for all methods. The quantile normalization is recommended when deconvolving microarray data. Therefore CIBERSORT's `QN` option corresponds to the `arrays` option of the `deconvolute`  and `deconvolute_cibersort` functions. 

Can you try the command 

`res = deconvolute(TPMs, "cibersort_abs", abs_method = "sig.score",  arrays=FALSE, perm=500)`

And check if the results are the same?

**Comment by RichardCorbett** [comment]
Thanks for the suggestion, @LorenzoMerotto!

In my analysis I am providing 1 sample's TPMs duplicated across 2 columns (with unique sample names) so I don't expect differences related to QN.     I tried running with your suggested command and still got the `0.14` estimate.

**Comment by FranziskaGoertler** [comment]
Your code did work? I try the same and get the following error: 

`cib <- deconvolute(Y_2[,1:100], "cibersort") `

and get the error: Error in (function (sig_matrix, mixture_file, perm = 0, QN = TRUE, saveLocation)  :  unused arguments (absolute = FALSE, abs_method = "sig.score")

What can I do here?
